tests/model_selection/randomized_scaffold_split.py

```diff
@@ -140,13 +185,86 @@ def test_seed_consistency_train_test_split(all_molecules):
 
 def test_seed_consistency_train_valid__test_split(all_molecules):
     train_set_1, valid_set_1, test_set_1 = randomized_scaffold_train_valid_test_split(
-        all_molecules, random_state=42
+        all_molecules, random_state=0
     )
 
     train_set_2, valid_set_2, test_set_2 = randomized_scaffold_train_valid_test_split(
-        all_molecules, random_state=42
+        all_molecules, random_state=0
     )
 
     assert train_set_1 == train_set_2
     assert valid_set_1 == valid_set_2
     assert test_set_1 == test_set_2
+
+
+def test_train_test_split_properly_splits_csk_with_ints(
+    smiles_ten_scaffolds,
+):
+    train_set, test_set = randomized_scaffold_train_test_split(
+        data=smiles_ten_scaffolds, train_size=7, test_size=3, use_csk=True
+    )
+    assert len(train_set) == 7
+    assert len(test_set) == 3
+
+
+def test_train_test_split_properly_splits_csk_with_floats(
+    smiles_ten_scaffolds,
+):
+    train_set, test_set = randomized_scaffold_train_test_split(
+        data=smiles_ten_scaffolds,
+        train_size=0.7,
+        test_size=0.3,
+        use_csk=True,
+    )
+    assert len(train_set) == 7
+    assert len(test_set) == 3
+
+
+def test_train_valid_test_split_properly_splits_csk_with_ints(
+    smiles_ten_scaffolds,
+):
+    train_set, valid_set, test_set = randomized_scaffold_train_valid_test_split(
+        data=smiles_ten_scaffolds,
+        train_size=7,
+        valid_size=2,
+        test_size=1,
+        use_csk=True,
+    )
+    assert len(train_set) == 7
+    assert len(valid_set) == 2
+    assert len(test_set) == 1
+
+
+def test_train_valid_test_split_properly_splits_csk_with_floats(
+    smiles_ten_scaffolds,
+):
+    train_set, valid_size, test_set = randomized_scaffold_train_valid_test_split(
+        data=smiles_ten_scaffolds,
+        train_size=0.7,
+        valid_size=0.2,
+        test_size=0.1,
+        use_csk=True,
+    )
+    assert len(train_set) == 7
+    assert len(valid_size) == 2
+    assert len(test_set) == 1
+
+
+def test_randomized_scaffold_train_test_split_return_indices(all_molecules):
+    train_size = 0.6
+    test_size = 0.4
+    result = randomized_scaffold_train_test_split(
+        all_molecules, train_size=train_size, test_size=test_size, return_indices=True
+    )
+
+    train_idxs, test_idxs = result
+
+    assert isinstance(train_idxs, list), "Train indices should be a list"
+    assert isinstance(test_idxs, list), "Test indices should be a list"
+
+    assert all(
+        isinstance(idx, int) for idx in train_idxs
+    ), "All train indices should be integers"
+    assert all(
+        isinstance(idx, int) for idx in test_idxs
+    ), "All test indices should be integers"
```

Comment:
No need for string errors here - if something fails, this should be obvious from the code. And it is IMO, since the code is nice and readable